pkg/dynamic/client.go

```diff
@@ -94,3 +111,86 @@ func (c *knDynamicClient) ListSourcesTypes() (*unstructured.UnstructuredList, er
 func (c knDynamicClient) RawClient() dynamic.Interface {
 	return c.client
 }
+
+// ListSources returns list of available sources COs
+func (c *knDynamicClient) ListSources(f *SourceListFilters) (*unstructured.UnstructuredList, error) {
+	var sourceList unstructured.UnstructuredList
+	options := metav1.ListOptions{}
+	sourceTypes, err := c.ListSourcesTypes()
```

Comment:
All CRUD with dynamic client work with `Unstructured` objects, in this case, its kind is set to `CustomResourceDefinitionList` (see `kn source list-types -oyaml`).

Reply:
I just wanted to say, for thus `UnstructuredList` we *know* the type (its `CustomeResourceDefinitionList` in advance, so could equally used a typed interface or marshall that list into the typed object.

For the source list its, different as it is a *heterogenous* list with different type of objects, so not really that fits well in the K8s model (a hint that this is not perfect is that you set the type of this arbitrarily to the type of the last item). The only thing they share is a common partial schema (duck type), so that fields from that partial schema can be printed out. Maybe you could introduced an object of this partial schema with the fields to printed out e.g `SourceDuck`, create a `SourceDuckList` type (which implements runtime.Object` that collect all the homogenous elements of the same object. 

Just a suggestion, we can iterate on this after this PR is merged (it gets a bit confusing otherwise).